src/client/datascience/data-viewing/data-wrangler/dataWrangler.ts

```diff
@@ -0,0 +1,461 @@
+            await super.loadWebview(process.cwd(), webviewPanel).catch(traceError);
+
+            const wantedPanels = this.configService.getSettings().dataWrangler.sidePanelSections;
+            this.postMessage(DataWranglerMessages.UpdateHistoryList, wantedPanels).ignoreErrors();
+
+            // Use Data Viewer logic to show initial data
+            const dataFrameInfo = await this.showInitialData(title);
+            this.sourceFile = dataFrameInfo.sourceFile;
+
+            this.historyList.push({
+                transformation: DataScience.dataWranglerImportTransformation(),
+                code: DataScience.dataWranglerImportCode().format(this.sourceFile ?? 'broken'),
+                variableName: 'df'
+            });
+            this.postMessage(DataWranglerMessages.UpdateHistoryList, this.historyList).ignoreErrors();
+        }
+    }
+
+    private dataWranglerDisposed() {
+        this._onDidDisposeDataWrangler.fire(this as IDataWrangler);
+    }
+
+    public async updateWithNewVariable(newVariableName: string) {
+        const notebook = (this.dataProvider as IJupyterVariableDataProvider).notebook;
+
+        // Generate a variable
+        const jupyterVariable = await this.kernelVariableProvider.getFullVariable(
+            {
+                name: newVariableName,
+                value: '',
+                supportsDataExplorer: true,
+                type: 'DataFrame',
+                size: 0,
+                shape: '',
+                count: 0,
+                truncated: true
+            },
+            notebook
+        );
+        const jupyterVariableDataProvider = await this.dataProviderFactory.create(jupyterVariable);
+        // Set dependencies for jupyterVariableDataProvider
+        jupyterVariableDataProvider.setDependencies(jupyterVariable, notebook);
+        // Get variable info
+        this.dataFrameInfoPromise = jupyterVariableDataProvider.getDataFrameInfo();
+        this.dataProvider = jupyterVariableDataProvider;
+        const dataFrameInfo = await this.dataFrameInfoPromise;
+        super.setTitle(`Data Wrangler`);
+
+        this.postMessage(DataViewerMessages.InitializeData, dataFrameInfo).ignoreErrors();
+    }
+
+    public async getHistoryItem(index: number) {
+        const variableName = this.historyList[index].variableName;
+        await this.updateWithNewVariable(variableName);
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    protected onMessage(message: string, payload: any) {
+        switch (message) {
+            case DataWranglerMessages.SubmitCommand:
+                this.handleCommand(payload).ignoreErrors();
+                break;
+
+            case DataWranglerMessages.RefreshDataWrangler:
+                this.refreshData().ignoreErrors();
+                break;
+
+            case InteractiveWindowMessages.LoadTmLanguageRequest:
+                void this.requestTmLanguage(payload);
```

Comment:
Shouldn't these be handled by the super.handleMessage?

Reply:
yep